### DISCUSSION TITLE
PDF Export!

### DISCUSSION THREAD

**Discussion by GypsyAcidQueen** [chuangcaleb/obsidian-fountain-editor#44] [Ideas]
Hi Caleb, first of all thanks so much for this wonderful plugin, it really works so well in the editor! Unfortunately I can't really use it without PDF export also reflecting the markdown! I need to be able to share the screenplay with others once it is written. Is there a workaround for this, or some way of implementing PDF export into the plugin? Thanks again for your work.

**Comment by chuangcaleb** [comment]
Eh sorry, didn't see this!

What do you mean by this?
>  I can't really use it without PDF export also reflecting the markdown

Currently (and personally), I'm using a follow-up external tool, [Better Fountain - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=piersdeseilligny.betterfountain) to export to PDF (which itself is based on [ifrost/afterwriting-labs: Post-processing for Fountain screenplays](https://github.com/ifrost/afterwriting-labs))—which works great for me. 

There's just a lot of settings and knobs to customising how the PDF Export would work, which would be retreading ground where others have already established. Let me know if you need further help?

**Reply by tendril-jeroen** [reply]
Hey @chuangcaleb ,

First of all, thanks for this amazing plugin! I use it every day and it’s been a huge help for my writing workflow.

I came across this discussion because I was also wondering if there are any plans for PDF/FDX export. I saw you mentioned using the VS Code Better Fountain extension for PDF export. Since Better Fountain (and afterwriting) is MIT-licensed, it should be possible to integrate its export functionality directly into this Obsidian plugin.

I’d be happy to put together a PR that adds this as a command, and potentially also hooks into Obsidian’s existing “Export to PDF” flow. That way, users like me wouldn’t need to switch to VS Code or another tool just to share their work. Everything could be done inside Obsidian.

Would you be open to that?

**Reply by tendril-jeroen** [reply]
Aha, I just read the following in the docs:

> This plugin should do [one thing well](https://en.wikipedia.org/wiki/Unix_philosophy), instead of many things half-baked.
> 
> “Lacking nothing” is different from “having everything”…
> 
> Should overly-extensive requirements arise, a brand new/additional tool should be created/used. The numerous [Longform - Obsidian Plugin](https://github.com/kevboh/longform) features should already cover almost all your additional needs, like [configurable manuscript compilation](https://github.com/kevboh/longform/blob/main/docs/COMPILE.md) and [word count](https://github.com/kevboh/longform/blob/main/docs/WORD_COUNTS.md)!
> 
> See [Synergistic Tools](https://obsidian-fountain-editor.chuangcaleb.com/docs/resources/synergistic-tools) for more recommendations!

I respect this approach. Seems it might be the better solution to create a new plugin for this then!

**Reply by chuangcaleb** [reply]
Hey @tendril-jeroen — so glad to hear you are using it every day, and find it beneficial! That makes the hours I spend, all the more worth it.

And glad to see you're really engaging with the workflow. I would like a PDF export command, but it's just so many moving parts and opens the door very wide. Thanks for checking out the docs, I spent so many hours on that too HAHA

I'm not sure if you could add a Longform hook or call custom script. If you get something to work, you could consider writing a really short tutorial on the docs/wiki site, so others like you could reference too! Please keep in touch?
